src/sentry/api/helpers/group_index.py

```diff
@@ -751,6 +752,56 @@ def update_groups(request, group_ids, projects, organization_id, search_fn):
                                         "status": GroupResolution.Status.resolved,
                                     }
                                 )
+                            else:
+                                # If we already know the `next` release in date based ordering
+                                # when clicking on `resolvedInNextRelease` because it is already
+                                # been released, there is no point in setting GroupResolution to
+                                # be of type in_next_release but rather in_release would suffice
+
+                                try:
+                                    # Get current release object from current_release_version
+                                    current_release_obj = Release.objects.get(
+                                        version=current_release_version,
+                                        organization_id=projects[0].organization_id,
+                                    )
+
+                                    date_order_q = Q(
+                                        date_added__gt=current_release_obj.date_added
+                                    ) | Q(
+                                        date_added=current_release_obj.date_added,
+                                        id__gt=current_release_obj.id,
+                                    )
```

Comment:
I'm wondering if this should be

`Q(date_added__gte=current_release_obj.date_added).exclude(id=current_release_obj.id)`

Mostly just trying to avoid the `or` here since it can make postgres skip index use. Since we sort by `sort, id` later the highest id will end up at the top. And in the extremely rare case that two releases have the exact same `date_added` and the newer one has a lower id and there are no newer releases we'll get the wrong id, but maybe that's ok?

Reply:
> the highest id will end up at the top
I think you mean will end up at the bottom .. This is ASC ordering

So in extreme case where they have exact same `date_added` if we add your suggested query. For example, 
```
release -> id = 1
release -> id = current_release_obj.id
release -> id = 4
```
We would get the previous release which is release of `id=1` as the next release, rather than release with `id=4`.
Does it ever happen that `date_added` can be exactly the same, or we okay just not caring about that particular case?

Reply:
Oops, misread the sort as `desc` instead of `asc`.

I just ran
```
select organization_id, date_added, count(*)
from sentry_release
group by organization_id, date_added
having count(*) > 1
```

I expected there not to be many rows, but we have 200k rows with the same `date_added`... definitely a surprise for me. So it might be worth leaving as-is for now, and if performance is an issue then resolve it later.

Reply:
Oh wow! I didn't actually think it would be that much !